example/rlottiePlayer/rlottiePlayer.cpp

```diff
@@ -0,0 +1,517 @@
+{
+    UNREFERENCED_PARAMETER(lParam);
+    switch (message)
+    {
+    case WM_INITDIALOG:
+        return (INT_PTR)TRUE;
+
+    case WM_COMMAND:
+        if (LOWORD(wParam) == IDOK || LOWORD(wParam) == IDCANCEL)
+        {
+            EndDialog(hDlg, LOWORD(wParam));
+            return (INT_PTR)TRUE;
+        }
+        break;
+    }
+    return (INT_PTR)FALSE;
+}
+
+void openJSONFileDialog(HWND hDlg)
+{
+    OPENFILENAME ofn;       // common dialog box structure
+    TCHAR szFile[260] = { 0 };       // if using TCHAR macros
+
+    // Initialize OPENFILENAME
+    ZeroMemory(&ofn, sizeof(ofn));
+    ofn.lStructSize = sizeof(ofn);
+    ofn.hwndOwner = hDlg;
+    ofn.lpstrFile = szFile;
+    ofn.nMaxFile = sizeof(szFile);
+    ofn.lpstrFilter = _T("JSON\0*.json\0");
+    ofn.nFilterIndex = 1;
+    ofn.lpstrFileTitle = NULL;
+    ofn.nMaxFileTitle = 0;
+    ofn.lpstrInitialDir = NULL;
+    ofn.Flags = OFN_PATHMUSTEXIST | OFN_FILEMUSTEXIST;
+
+    if (GetOpenFileName(&ofn))
+    {
+        SetWindowText(hTextFileToBeOpened, ofn.lpstrFile);
+        // LPWSTR(w_char*) -> LPSTR(char*)
+        USES_CONVERSION;
+        LPSTR path = W2A(ofn.lpstrFile);
+
+        setAnimation(path, BMP_MAX_LEN, BMP_MAX_LEN);
+        // init play slider
+        SendMessage(hSliderPlay, TBM_SETRANGE, FALSE, MAKELPARAM(0, getTotalFrame()));
+        SendMessage(hSliderPlay, TBM_SETPOS, TRUE, 0);
+        renderAnimation(0);
+    }
+}
+
+void draw(HDC hdc)
+{
+    Graphics gf(hdc);
+    int half_interval = UI_INTERVAL / 2;
+
+    // background
+    SolidBrush brush(backColor);
+    int back_y = half_interval + BTN_HEIGHT;
+    int back_height = back_y + BMP_MAX_LEN + UI_INTERVAL;
+    if (isBackgroundChanged)
+    {
+        isBackgroundChanged = false;
+        gf.FillRectangle(&brush, 0, back_y, WND_WIDTH, back_height);
+    }
+
+    // image borderline
+    Pen pen(borderColor);
+    gf.DrawRectangle(&pen, anim.x - half_interval, anim.y - half_interval, anim.width + half_interval * 2, anim.height + half_interval * 2);
+
+    // image
+    if (anim.image != NULL)
+    {
+        gf.DrawImage(anim.image, anim.x, anim.y, anim.width, anim.height);
+    }
+}
+
+Bitmap* CreateBitmap(void* data, unsigned int width, unsigned int height)
+{
+    BITMAPINFO Info;
+    memset(&Info, 0, sizeof(Info));
+
+    Info.bmiHeader.biSize = sizeof(BITMAPINFOHEADER);
+    Info.bmiHeader.biWidth = 500;
+    Info.bmiHeader.biHeight = 500;
+    Info.bmiHeader.biPlanes = 1;
+    Info.bmiHeader.biBitCount = 32;
+    Info.bmiHeader.biCompression = BI_RGB;
+    Info.bmiHeader.biSizeImage = 0;  //(((32 * width + 31) & ~31) / 8) * height;
+
+    return new Gdiplus::Bitmap(&Info, data);
+}
+
+void renderAnimation(UINT frameNum)
+{
+    if (isAnimNULL()) return;
+    if (anim.image != NULL) delete anim.image;
+
+    curFrame = frameNum % getTotalFrame();
+
+    // render
+    UINT* resRender = renderRLottieAnimation(curFrame);
+    anim.image = CreateBitmap(resRender, BMP_MAX_LEN, BMP_MAX_LEN);
```

Comment:
rather than fixed size (500), You can dynamically decide the animation size by querying default size of animation.
After Animation::loadfromFile(..);
You can use this: Animation::size();

This feedback is just for you information, I don't argue you to do this.

Reply:
I revised it as you gave me feedback.